noxfiles/git_nox.py

```diff
@@ -145,12 +146,16 @@ def increment_tag(
         version=version_number, tag_type=tag_type.value
     )
     # find our latest existing tag for this version/type
-    latest_tag = next(
+    sorted_tag_matches = sorted(
         (
-            re.fullmatch(version_branch_tag_pattern, tag.name)
+            tag.name
             for tag in all_tags
             if re.fullmatch(version_branch_tag_pattern, tag.name)
         ),
+        reverse=True,
+    )
+    latest_tag = next(
+        (re.fullmatch(version_branch_tag_pattern, tag) for tag in sorted_tag_matches),
```

Comment:
this `re.fullmatch` shouldn't be needed again since we've still got that conditional in the initial sort on L152/3, right? shouldn't be a problem, i think it's just redundant if i'm reading correctly.

Reply:
I don't think it is, but I figured it was safest to keep the output of `latest_tag` consistent. Will take a closer look 👍🏽

Reply:
cool, really not a big deal either way - just think if we can remove it, it'd make things a bit more readable!

Reply:
Agreed! I tried a few things and it still felt a bit wonky unfortunately with how we increment. The closest possible improvement that I could think of without some risky/fuzzy string matching is in https://github.com/ethyca/fides/pull/3405/commits/76fc550af2451900f6a22d85dcf8a3b7879fb755 if you want to let me know what you think. Happy to go any route you see fit!

Reply:
ohh right i get it now, sorry this was an unnecessary nit from me. i'm good with where you landed!